packages/request-client.js/test/api/payment-network/erc20/proxy-contract.test.ts

```diff
@@ -0,0 +1,94 @@
+import { AdvancedLogicTypes, RequestLogicTypes } from '@requestnetwork/types';
+import ERC20ProxyContract from '../../../../src/api/payment-network/erc20/proxy-contract';
+
+import * as chai from 'chai';
+import * as spies from 'chai-spies';
+import 'mocha';
+
+const expect = chai.expect;
+chai.use(spies);
+const sandbox = chai.spy.sandbox();
+
+let erc20ProxyContract: ERC20ProxyContract;
+
+const mockAdvancedLogic: AdvancedLogicTypes.IAdvancedLogic = {
+  applyActionToExtensions(): any {
+    return;
+  },
+  extensions: {
+    proxyContractErc20: {
+      createAddPaymentAddressAction(): any {
+        return;
+      },
+      createAddRefundAddressAction(): any {
+        return;
+      },
+      createCreationAction(): any {
+        return;
+      },
+    },
+  },
+};
+
+/* tslint:disable:no-unused-expression */
+describe('api/erc20/proxy-contract', () => {
+  beforeEach(() => {
+    sandbox.restore();
+    erc20ProxyContract = new ERC20ProxyContract({ advancedLogic: mockAdvancedLogic });
+  });
+
+  it('can createExtensionsDataForCreation', async () => {
+    const spy = sandbox.on(mockAdvancedLogic.extensions.proxyContractErc20, 'createCreationAction');
+
+    erc20ProxyContract.createExtensionsDataForCreation({
+      paymentAddress: 'ethereum address',
+      salt: 'ea3bc7caf64110ca',
+    });
+
+    expect(spy).to.have.been.called.once;
```

Comment:
this doesn't test much. We should at least check the argument given to the funcion called.

Reply:
The problem is for all PN tests, so I created this task:
https://requestnetwork.atlassian.net/secure/RapidBoard.jspa?rapidView=22&projectKey=PROT&view=planning&selectedIssue=PROT-1056

Also chai seems to not support `to.be.called.with()` with object with parameter, it seems we need to use sinon package instead.

Reply:
Discussed IRL. 
chai support it but the error message is not helping to understand what happened.